### PR TITLE
Add support for ClassLoader config

### LINKED ISSUE #11962: quarkus:dev - Lift external dependency into application-classloader
**Description**
External libraries might be keeping state in some statics that does not play nicely with quarkus dev-mode reloads. This is because everything external to the main maven/gradle project goes into the application base classloader that stays the same between dev-mode runs/reloads.
With the gradle plugin not supporting composite builds and also in general, it would be nice to somehow tell quarkus that certain external dependencies/libs should be lifted up into the application-classloader (which is discarded and reloaded during dev-mode runs).

cc @stuartwdouglas: I asked for this during quarkus insights

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #11962

### CODE REVIEW

**Review by geoand — APPROVED** [review]
👍🏼

File: docs/src/main/asciidoc/class-loading-reference.adoc [hunk 1]
```diff
@@ -191,3 +172,39 @@ file). Simply add an `excludedArtifacts` section as shown below:
...
+
+
+`quarkus.class-loading.parent-first-artifacts`::
+A comma separated list of artifacts (in the groupId:artifactId format) that will be loaded parent first. Note that if you
+make a library parent first all its dependencies should generally also be parent first.
+
+`quarkus.class-loading.reloadable-artifacts`::
+A comma separated list of artifacts (in the groupId:artifactId format) that will be loaded in the runtime ClassLoader. This
+means they will be recreated on every hot reload. This can be used to work around issues with 3rd party libraries that
+save static state. Note that if you use this any library that depends on the listed libraries will also need to be reloadable.
```
Comment (machi1990) [inline docs/src/main/asciidoc/class-loading-reference.adoc:190]: You could use the generated `quarkus-class-loading-configuration-class-loading-config.adoc` file.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java [hunk 2]
```diff
@@ -158,7 +157,8 @@ private synchronized void processCpElement(AppArtifact artifact, Consumer<ClassP
 
     private void addCpElement(QuarkusClassLoader.Builder builder, AppArtifact dep, ClassPathElement element) {
         final AppArtifactKey key = dep.getKey();
-        if (appModel.getParentFirstArtifacts().contains(key)) {
+        if (appModel.getParentFirstArtifacts().contains(key)
```
Comment (aloubyansky) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java:160]: AppArtifact.getKey() returns a key that includes groupId:artifactId:classifier:type, while the classloading config expects groupId:artifactId. The same for reloadableArtifacts below.

Comment (stuartwdouglas) [reply independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/CuratedApplication.java:160]: That is fine though, as it uses the String[] constructor for AppArtifactKey that will fill in the missing info if only GA is provided.

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**